snapcraft/internal/pluginhandler/_mangling.py

```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+import os
+import re
+from typing import FrozenSet, List
+
+from snapcraft import file_utils
+from snapcraft.internal import elf, repo
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_dynamic_linker(library_list: List[str]) -> str:
+    """Return the dynamic linker from library_list."""
+    regex = re.compile(r'(?P<dynamic_linker>ld-[\d.]+.so)$')
+
+    for library in library_list:
+        m = regex.search(os.path.basename(library))
+        if m:
+            return m.group('dynamic_linker')
+
+    raise NotImplementedError(
```

Comment:
I don't think this should be a NotImplementedError.
A more snapcraft-specific exception would be better.

Reply:
Agreed, although I think this is the one @sergiusens specifically did not want to make friendly. A RuntimeError would be better-suited, if needed.

Reply:
`RuntimeError` it shall be, this is not anything a user can fix and a horrible thing if it happens so the error should look just as grotesque for us to fix fast.